plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -3002,9 +3020,13 @@ else if (tableChangeInfo instanceof UnknownTableChange) {
             }
         }
 
+        if (!dependsOnTableFunctions.isEmpty()) {
+            return new MaterializedViewFreshness(UNKNOWN, refreshTime);
+        }
         Instant lastFreshTime = firstTableChange
                 .map(Instant::ofEpochMilli)
-                .orElse(refreshTime);
+                .or(() -> refreshTime)
+                .orElseThrow();
```

Comment:
We're not supposed to throw here.
Document why throwing is impossible here.

Reply:
making `currentSnapshot` optional removes the need for this orElseThrow

Reply:
reworked a little to avoid this